src/components/operators.jl

```diff
@@ -213,8 +213,10 @@ function parse_unary_colon(ps::ParseState, op::EXPR)
     elseif closer(ps)
         ret = op
     else
+        prev_errored = ps.errored
         arg = @precedence ps 20 parse_expression(ps)
         if typof(arg) === InvisBrackets && length(arg.args) == 3 && typof(arg.args[2]) === ErrorToken && errorof(arg.args[2]) === UnexpectedAssignmentOp
+            ps.errored = prev_errored
```

Comment:
Here's the wonky special case where we need to reset the error state.

It makes me worry that the whole approach is a bit fragile, but it's the only instance I found.